crates/core_simd/examples/nbody.rs

```diff
@@ -0,0 +1,192 @@
+            1.62824170038242295e-03 * DAYS_PER_YEAR,
+            -9.51592254519715870e-05 * DAYS_PER_YEAR,
+            0.,
+        ]),
+        mass: 5.15138902046611451e-05 * SOLAR_MASS,
+    },
+];
+
+pub fn offset_momentum(bodies: &mut [Body; N_BODIES]) {
+    let (sun, rest) = bodies.split_at_mut(1);
+    let sun = &mut sun[0];
+    for body in rest {
+        let m_ratio = body.mass / SOLAR_MASS;
+        sun.v -= body.v * m_ratio;
+    }
+}
+
+pub fn energy(bodies: &[Body; N_BODIES]) -> f64 {
+    let mut e = 0.;
+    for i in 0..N_BODIES {
+        let bi = &bodies[i];
+        e += bi.mass * (bi.v * bi.v).horizontal_sum() * 0.5;
+        for bj in bodies.iter().take(N_BODIES).skip(i + 1) {
+            let dx = bi.x - bj.x;
+            e -= bi.mass * bj.mass / (dx * dx).horizontal_sum().sqrt()
+        }
+    }
+    e
+}
+
+pub fn advance(bodies: &mut [Body; N_BODIES], dt: f64) {
+    const N: usize = N_BODIES * (N_BODIES - 1) / 2;
+
+    // compute distance between bodies:
+    let mut r = [f64x4::splat(0.); N];
+    {
+        let mut i = 0;
+        for j in 0..N_BODIES {
+            for k in j + 1..N_BODIES {
+                r[i] = bodies[j].x - bodies[k].x;
+                i += 1;
+            }
+        }
+    }
+
+    let mut mag = [0.0; N];
+    let mut i = 0;
+    while i < N {
+        let d2s = f64x2::from_array([(r[i] * r[i]).horizontal_sum(), (r[i + 1] * r[i + 1]).horizontal_sum()]);
+        let dmags = f64x2::splat(dt) / (d2s * d2s.sqrt());
+        dmags.write_to_slice_unaligned(&mut mag[i..]);
+        i += 2;
+    }
+
+    i = 0;
+    for j in 0..N_BODIES {
+        for k in j + 1..N_BODIES {
+            let f = r[i] * mag[i];
+            bodies[j].v -= f * bodies[k].mass;
+            bodies[k].v += f * bodies[j].mass;
+            i += 1
+        }
+    }
+    for body in bodies {
+        body.x += dt * body.v
+    }
+}
+
+pub fn run_k<K>(n: usize, k: K) -> (f64, f64)
+where
+    K: Fn(&mut [Body; N_BODIES], f64),
+{
+    let mut bodies = BODIES;
+    offset_momentum(&mut bodies);
+    let energy_before = energy(&bodies);
+    for _ in 0..n {
+        k(&mut bodies, 0.01);
+    }
+    let energy_after = energy(&bodies);
+
+    (energy_before, energy_after)
+}
+
+pub fn run(n: usize) -> (f64, f64) {
+    run_k(n, advance)
+}
+
+const OUTPUT: Vec<f64> = vec![-0.169075164, -0.169087605];
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn test() {
+        let mut out: Vec<u8> = Vec::new();
+        run(&mut out, 1000, 0);
+        for &(size, a_e, b_e) in crate::RESULTS {
+            let (a, b) = super::run(size);
+            assert_eq!(format!("{:.9}", a), a_e);
+            assert_eq!(format!("{:.9}", b), b_e);
+        }
+    }
+}
+fn main() {
+    //let n: usize = std::env::args()
+    //.nth(1)
+    //.expect("need one arg")
+    //.parse()
+    //.expect("argument should be a usize");
+    //run(&mut std::io::stdout(), n, alg);
+    println!("{:?}", run_k::<10>(10, 10));
```

Comment:
This turbofish causes a disagreement between this constant value and the previous definition of <K> as `Fn(&mut [Body; N_BODIES], f64)`, as it is a constant value and not a function. You may have meant to use just `run`?

It would also probably be fine if your function accepted a mutable slice, `&mut [Body]`, currently it's expecting specifically `Fn<(&mut [Body; 5], f64)>`.

Reply:
Oh right - `rustc` called me out on this too. I'll change it to `run(10)` and that should do it.